src/tasks/minions/camdozaalActivity/camdozaalMiningActivity.ts

```diff
@@ -0,0 +1,127 @@
+import { ActivityTaskOptionsWithQuantity } from '../../../lib/types/minions';
+import { roll, skillingPetDropRate } from '../../../lib/util';
+import { handleTripFinish } from '../../../lib/util/handleTripFinish';
+import { makeBankImage } from '../../../lib/util/makeBankImage';
+
+// Custom gem loot table for barronite rocks
+const gemTable = new LootTable()
+	.add('Uncut sapphire', 0, 70)
+	.add('Uncut sapphire', 1, 32)
+	.add('Uncut emerald', 1, 16)
+	.add('Uncut ruby', 1, 8)
+	.add('Uncut diamond', 1, 8);
+
+// Barronite rock loot table
+const barroniteTable = new LootTable().add('Barronite shards', [4, 6], 76).add('Barronite deposit', 1, 24);
+
+export const camdozaalMiningTask: MinionTask = {
+	type: 'CamdozaalMining',
+	async run(data: ActivityTaskOptionsWithQuantity) {
+		let { quantity, userID, channelID, duration } = data;
+		const user = await mUserFetch(userID);
+		const currentMiningLevel = user.skillLevel(SkillsEnum.Mining);
+
+		let barroniteShardMined = 0;
+		let barroniteDepositMined = 0;
+
+		// Count loot received during trip
+		const loot = new Bank();
+
+		for (let i = 0; i < quantity; i++) {
+			if (roll(256)) {
+				loot.add(gemTable.roll());
+			} else {
+				const barroniteMined = barroniteTable.roll();
+
+				if (barroniteMined.has('Barronite shards')) {
+					barroniteShardMined++;
+					loot.add(barroniteMined);
+					continue;
+				}
+				if (barroniteMined.has('Barronite deposit')) {
+					barroniteDepositMined++;
+					loot.add(barroniteMined);
+					continue;
+				}
+			}
+		}
+
+		// Add up the xp from the trip
+		let miningXpReceived = barroniteShardMined * 16 + barroniteDepositMined * 32;
+		let bonusXP = 0;
+
+		// If user has the entire prospector outfit, give an extra 2.5% xp bonus
+		if (
+			user.gear.skilling.hasEquipped(
+				Object.keys(Mining.prospectorItems).map(i => parseInt(i)),
+				true
+			)
+		) {
+			const amountToAdd = Math.floor(miningXpReceived * (2.5 / 100));
+			miningXpReceived += amountToAdd;
+			bonusXP += amountToAdd;
+		} else {
+			// For each prospector item, check if they have it, give its' XP boost
+			for (const [itemID, bonus] of Object.entries(Mining.prospectorItems)) {
+				if (user.hasEquipped(parseInt(itemID))) {
+					const amountToAdd = Math.floor(miningXpReceived * (bonus / 100));
+					miningXpReceived += amountToAdd;
+					bonusXP += amountToAdd;
+				}
+			}
+		}
+
+		// Add xp to user
+		const xpRes = await user.addXP({
+			skillName: SkillsEnum.Mining,
+			amount: miningXpReceived,
+			duration,
+			source: 'CamdozaalMining'
+		});
+
+		// Trip finish message
+		let str = `${user}, ${user.minionName} finished mining in Camdozzal! ${xpRes}`;
+		if (bonusXP > 0) {
+			str += `\n\n**Bonus XP:** ${bonusXP.toLocaleString()}`;
+		}
+
+		// Add clue scrolls
+		const clueScrollChance = 257_770;
+		addSkillingClueToLoot(user, SkillsEnum.Mining, quantity, clueScrollChance, loot);
+
+		// Rock golem roll
+		const totalBarroniteMined = barroniteShardMined + barroniteShardMined;
```

Comment:
You could just use *2 if you intended it to double

Reply:
Shards drop at rate of 76/100 and deposits drop at a rate of 24/100. But the pet is rolled after each barronite rock mined.

Reply:
My bad, I see what you mean now. It should read: 
const totalBarroniteMined = barroniteShardMined + barroniteDepositMined;

Thank you